merlin/systems/dag/ops/pytorch.py

```diff
@@ -16,8 +16,11 @@
 import os
 
 import numpy as np
-import torch
-import torch.utils.dlpack
+
+try:
+    import torch
+except ImportError:
+    torch = None
```

Comment:
Updated. There is a side-effect in the `merlin.core.compat.tensorflow` that does more than importing tensorflow, which I'm not sure about - the `configure_tensorflow` function. It may be worth looking at making the device configuration something that is called explictly instead of called as a result of the import of that module.

Reply:
There are arguments for both sides here. We have had customer complaints before about TF taking up more space than required. The issue manifests itself as OOM error, and this takes a while to investigate given the back and forth required with the customer. So we have to go through this whole thing that explains that it is required that you use the configure_tensorflow method to clamp down TF gpu memory reservation. However this ends up confusing the customer, so we wanted to handle that for the user.

Reply:
We used to do it explicitly, but it was a giant pain to remember to do it everywhere and if you didn't remember to do it before importing `tensorflow` then things would break. I'm honestly not sure that doing it this way is entirely better because you can't really set the parameters anymore, but 🤷🏻 six of one, half dozen of the other.